python-backend/app/api/variances/resources/variance_resource.py

```diff
@@ -18,22 +19,47 @@ class VarianceResource(Resource, UserMixin, ErrorMixin):
         description='Get a list of variances.',
         params={
             'page': f'The page number of paginated records to return. Default: {PAGE_DEFAULT}',
-            'per_page': f'The number of records to return per page. Default: {PER_PAGE_DEFAULT}'
+            'per_page': f'The number of records to return per page. Default: {PER_PAGE_DEFAULT}',
+            'variance_application_status_code':
+            'Comma-separated list of code statuses to include in results. Default: All status codes.',
         })
     @requires_any_of([MINE_VIEW])
     @api.marshal_with(PAGINATED_VARIANCE_LIST, code=200)
     def get(self):
-        paginated_variances, pagination_details = apply_pagination(
-            Variance.query, request.args.get('page', PAGE_DEFAULT, type=int),
-            request.args.get('per_page', PER_PAGE_DEFAULT, type=int))
+        records, pagination_details = self._apply_filters_and_pagination(
+            page_number=request.args.get('page', PAGE_DEFAULT, type=int),
+            page_size=request.args.get('per_page', PER_PAGE_DEFAULT, type=int),
+            application_status=request.args.get('variance_application_status_code', type=str))
 
-        if not paginated_variances:
+        if not records:
             raise BadRequest('Unable to fetch variances.')
 
         return {
-            'records': paginated_variances.all(),
+            'records': records.all(),
             'current_page': pagination_details.page_number,
             'total_pages': pagination_details.num_pages,
             'items_per_page': pagination_details.page_size,
             'total': pagination_details.total_results,
         }
+
+
+    def _apply_filters_and_pagination(self,
+                                      page_number=PAGE_DEFAULT,
+                                      page_size=PER_PAGE_DEFAULT,
+                                      application_status=None):
+        status_filter_values = list(map(
+            lambda x: x.variance_application_status_code,
+            VarianceApplicationStatusCode.active()))
+
+        if application_status is not None:
+            status_filter_values = application_status.split(',')
```

Comment:
Is it possible to have multiple application statuses or is this used in searching for all variances in a list of application statuses?

Reply:
This searches across all variances and only returns the ones with the requested application status(es). If no statuses were provided, it filters with all statuses -- ie. no variances will be filtered out.